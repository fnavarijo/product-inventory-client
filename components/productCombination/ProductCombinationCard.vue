<template>
  <NuxtLink :to="`/catalog/${id}`" class="group">
    <div
      class="w-full pt-cardRatio relative rounded-md bg-cover"
      :style="{ backgroundImage: `url(${image})` }"
    >
      <div
        class="absolute top-0 bottom-0 left-0 right-0 bg-dark-gray-700 rounded-md opacity-0 transition ease-linear duration-100 group-hover:opacity-25"
      />
      <div
        class="absolute -bottom-16 bg-primary-600 text-white w-full font-bold transition-bottom ease-in-expo duration-500 text-center p-5 opacity-0 invisible rounded-b-md uppercase group-hover:bottom-0 group-hover:opacity-100 group-hover:visible"
      >
        Ver
      </div>
    </div>
  </NuxtLink>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'ProductCombinationCard',
  props: {
    id: {
      type: String,
      required: true,
    },
    image: {
      type: String,
      required: true,
    },
  },
});
</script>
