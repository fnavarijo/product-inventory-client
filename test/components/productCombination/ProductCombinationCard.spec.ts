import { mount, RouterLinkStub } from '@vue/test-utils';

import ProductCombinationCard from '@/components/productCombination/ProductCombinationCard.vue';

describe('ProductCombinationCard component', () => {
  it('should render card with style', () => {
    const wrapper = mount(ProductCombinationCard, {
      propsData: {
        id: '123456',
        image: 'http://localhost/image.jpg',
      },
      stubs: {
        NuxtLink: RouterLinkStub,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
