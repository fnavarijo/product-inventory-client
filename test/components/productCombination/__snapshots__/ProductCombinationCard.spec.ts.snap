// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCombinationCard component should render card with style 1`] = `
<a class="group">
  <div class="w-full pt-cardRatio relative rounded-md bg-cover" style="background-image: url(http://localhost/image.jpg);">
    <div class="absolute top-0 bottom-0 left-0 right-0 bg-dark-gray-700 rounded-md opacity-0 transition ease-linear duration-100 group-hover:opacity-25"></div>
    <div class="absolute -bottom-16 bg-primary-600 text-white w-full font-bold transition-bottom ease-in-expo duration-500 text-center p-5 opacity-0 invisible rounded-b-md uppercase group-hover:bottom-0 group-hover:opacity-100 group-hover:visible">
      Ver
    </div>
  </div>
</a>
`;
